src/app/shared/header.component.ts

```diff
@@ -0,0 +1,72 @@
+const headerLayout: HTMLTemplateElement = document.createElement('template');
+headerLayout.innerHTML = `
+	<header class="d-flex justify-content-between bg-secondary px-3 py-2">
+		<div class="d-flex align-items-center">
+			<button id="prev-toolbar" type="button" class="btn btn-icon btn-inverse btn-secondary" data-title-i18n="previous">
+				<span class="visually-hidden" data-i18n="previous"></span>
+				<app-icon data-name="Form_Chevron_left"></app-icon>
+			</button>
+			<span id="title-app" class="d-flex gap-1 align-items-center fs-3 fw-bold text-white">
+				<app-icon data-size="2rem" data-name="Accessibility"></app-icon>
+				<span data-i18n="mainTitle"></span>
+				<span class="text-primary">+</span>
+			</span>
+		</div>
+		<button id="close-toolbar" type="button" class="btn btn-icon btn-inverse btn-primary" data-title-i18n="close">
+				<span class="visually-hidden" data-i18n="close"></span>
+				<app-icon data-name="Form_Chevron_right"></app-icon>
+		</button>
+	</header>
+`;
+
+class HeaderComponent extends HTMLElement {
+	closeBtn: HTMLElement | null = null;
+	prevBtn: HTMLElement | null = null;
+	titleApp: HTMLElement | null = null;
+	mode = 'primary';
+
+	constructor() {
+		super();
+
+		this.mode = this.dataset.mode || this.mode;
+
+		this.appendChild(headerLayout.content.cloneNode(true));
+	}
+
+	connectedCallback(): void {
+		this.closeBtn = this.querySelector('#close-toolbar');
+		this.prevBtn = this.querySelector('#prev-toolbar');
+		this.titleApp = this.querySelector('#title-app');
```

Comment:
Quand tu cibles un `id` (qui est donc censé être unique), `getElementById()` est plus performant — mais nécessite de remonter à la racine (à vérifier) : `this?.shadowRoot?.getElementById('close-toolbar');`, mais `this.getElementById('close-toolbar')` fonctionne peut-être directement…

Reply:
Merci pour tes retours ! J'ai testé et il faut bien remonter à la racine pour utiliser getElementById().